packages/core/src/rules/common/spec.ts

```diff
@@ -114,17 +115,20 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
           propValue = resolve(propValue).node;
         }
 
-        if (propSchema.enum) {
-          if (!propSchema.enum.includes(propValue)) {
-            report({
-              location: propLocation,
-              message: `\`${propName}\` can be one of the following only: ${propSchema.enum
-                .map((i) => `"${i}"`)
-                .join(', ')}.`,
-              from: refLocation,
-              suggest: getSuggest(propValue, propSchema.enum),
-            });
+        if (propSchema.items && propSchema.items?.enum && Array.isArray(propValue)) {
+          for (const item of propValue) {
+            validateSchemaEnumType(propSchema.items?.enum, item, propName, refLocation, {
+              report,
+              location,
```

Comment:
Can we change the loop to regular `for` and use the index for location here:

```suggestion
              location: location.child([propName, i])
```

This way it will highlight a specific array item that is invalid vs highlighting the whole array

Reply:
Yes, Thank you